src/gitHub/GitHubBranchListStep.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem, nonNullProp } from "@microsoft/vscode-azext-utils";
+import { QuickPickItem } from "vscode";
+import { loadMoreQp } from "../constants";
+import { localize } from "../utils/localize";
+import { IGitHubContext } from "./IGitHubContext";
+import { Branches, GetBranchesParams, getBranches } from "./getBranches";
+
+export class GitHubBranchListStep extends AzureWizardPromptStep<IGitHubContext> {
+    picks: IAzureQuickPickItem<string>[];
+
+    public async prompt(context: IGitHubContext): Promise<void> {
+        // We always want fresh picks before prompting in case the user has pressed the back button
+        this.picks = [];
+
+        const placeHolder: string = localize('gitHubBranch', 'Choose a GitHub branch');
```

Comment:
Aren't GitHub branches remote branches though 🙃

Reply:
I'd probably been more likely to just opt them all to be generic (i.e. organization, repository, branch) since I think that GitHub is pretty much implied once you start "Connecting to Github...". 

My preference is probably none, all, some in terms of the qualifier so do as you will.

Reply:
I'll do none then :)